tests/Doctrine/Tests/DBAL/Platforms/AbstractPlatformTestCase.php

```diff
@@ -629,17 +629,30 @@ public function testKeywordList() : void
         self::assertTrue($keywordList->isKeyword('table'));
     }
 
+    public function getAutoQuotedIdentifiers() {
+        return [
+            ['create'],
+            ['select'],
+            ['a name with spaces'],
+            ['a name with/slash'],
+            ['name-with-dash'],
+        ];
+    }
+
     /**
      * @group DBAL-374
+     * @group DBAL-2979
+     * @dataProvider getAutoQuotedIdentifiers
      */
-    public function testQuotedColumnInPrimaryKeyPropagation() : void
+    public function testAutoQuotedColumnInPrimaryKeyPropagation(string $identifier) : void
     {
         $table = new Table('`quoted`');
-        $table->addColumn('create', 'string', ['length' => 255]);
-        $table->setPrimaryKey(['create']);
+        $table->addColumn($identifier, 'string', ['length' => 255]);
+        $table->setPrimaryKey([$identifier]);
 
+        $expected = preg_replace('/create/', $identifier, $this->getQuotedColumnInPrimaryKeySQL());
```

Comment:
Not a huge fan of this… I would add a new argument defaulting to `create` to `getQuotedColumnInPrimaryKeySQL`.